librz/magic/magic-test.c

```diff
@@ -0,0 +1,1372 @@
+
+	if (ml->type_operator == '&')
+		value &= (uint32_t)ml->type_operand;
+	else if (ml->type_operator == '-')
+		value -= (uint32_t)ml->type_operand;
+	else if (ml->type_operator == '+')
+		value += (uint32_t)ml->type_operand;
+	else if (ml->type_operator != ' ')
+		return (-1);
+
+	result = magic_test_unsigned(ml, value, (uint32_t)ml->test_unsigned);
+	if (result == !ml->test_not && ml->result != NULL) {
+		t = value;
+		switch (ml->type) {
+		case MAGIC_TYPE_LDATE:
+		case MAGIC_TYPE_LELDATE:
+		case MAGIC_TYPE_BELDATE:
+			ctime_r(&t, s);
+			break;
+		default:
+			asctime_r(gmtime(&t), s);
+			break;
+		}
+		s[strcspn(s, "\n")] = '\0';
+		magic_add_result(ms, ml, "%s", s);
+		ms->offset += sizeof value;
+	}
+	return (result);
+}
+
+static int magic_test_type_uqdate(RzMagicLine *ml, RzMagicState *ms) {
+	uint64_t value;
+	int result;
+	time_t t;
+	char s[64];
+
+	if (magic_copy_from(ms, -1, &value, sizeof value) != 0)
+		return (0);
+	if (ml->type == MAGIC_TYPE_UBEQDATE ||
+		ml->type == MAGIC_TYPE_UBEQLDATE)
+		value = be64toh(value);
+	if (ml->type == MAGIC_TYPE_ULEQDATE ||
+		ml->type == MAGIC_TYPE_ULEQLDATE)
+		value = le64toh(value);
+
+	if (ml->type_operator == '&')
+		value &= (uint64_t)ml->type_operand;
+	else if (ml->type_operator == '-')
+		value -= (uint64_t)ml->type_operand;
+	else if (ml->type_operator == '+')
+		value += (uint64_t)ml->type_operand;
+	else if (ml->type_operator != ' ')
+		return (-1);
+
+	result = magic_test_unsigned(ml, value, (uint64_t)ml->test_unsigned);
+	if (result == !ml->test_not && ml->result != NULL) {
+		t = value;
+		switch (ml->type) {
+		case MAGIC_TYPE_UQLDATE:
+		case MAGIC_TYPE_ULEQLDATE:
+		case MAGIC_TYPE_UBEQLDATE:
+			ctime_r(&t, s);
+			break;
+		default:
+			asctime_r(gmtime(&t), s);
+			break;
+		}
+		s[strcspn(s, "\n")] = '\0';
+		magic_add_result(ms, ml, "%s", s);
+		ms->offset += sizeof value;
+	}
+	return (result);
+}
+
+static int magic_test_type_bestring16(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_lestring16(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_melong(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_medate(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_meldate(__unused RzMagicLine *ml,
+	__unused RzMagicState *ms) {
+	return (-2);
+}
+
+static int magic_test_type_regex(RzMagicLine *ml, RzMagicState *ms) {
+	const char *cp;
+	RzRegex *re;
+	int flags = 0, sflag = 0;
+
+	cp = &ml->type_string[(sizeof "regex") - 1];
+	if (*cp != '\0') {
+		if (*cp != '/')
+			return (-1);
+		cp++;
+		for (; *cp != '\0'; cp++) {
+			switch (*cp) {
+			case 's':
+				sflag = 1;
+				break;
+			case 'c':
+				flags |= RZ_REGEX_CASELESS;
+				break;
+			default:
+				return (-1);
+			}
+		}
+	}
+	re = rz_regex_new(ml->test_string, RZ_REGEX_EXTENDED | flags, 0, NULL);
```

Comment:
Sorry, you actually need to pass `RZ_REGEX_MULTILINE`. I remembered the API incorrectly.

```suggestion
	re = rz_regex_new(ml->test_string, RZ_REGEX_EXTENDED | RZ_REGEX_MULTILINE | flags, 0, NULL);
```